### ISSUE TITLE
Trigger CA1305 when string interpolation is converted to string.Format

### ISSUE THREAD

**Issue by NN---** [dotnet/roslyn#50057]
**Brief description:**

These lines are exactly the same but you have CA1305 on the second line while you don't on the first:

```cs
var p2 = $"{Environment.ProcessId}"; // No CA1305
var p3 = string.Format("{0}", Environment.ProcessId); // CA1305
```

String interpolation doesn't receive any IFormatProvider which means it formats according to the current culture the same way string.Format does.

**Languages applicable:**

C#

**Code example that the analyzer should report:**

Any string interpolation

**Additional information:**

Any more additional information you would like to add.

**Documentation requirements:**

When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](https://github.com/dotnet/roslyn/blob/master/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md).

**Comment by Youssef1313** [comment]
@mavasani This needs to be moved to dotnet/roslyn-analyzers repository as it's related to a CAxxxx rule.

**Comment by NN---** [comment]
Sorry, wrong repo.
Thanks
